src/utils.js

```diff
@@ -1,37 +1,51 @@
 export const geocodeByAddress = (address, callback) => {
-  const geocoder = new google.maps.Geocoder()
-  const OK = google.maps.GeocoderStatus.OK
+  return new Promise((resolve, reject) => {
+    const geocoder = new google.maps.Geocoder()
+    const OK = google.maps.GeocoderStatus.OK
 
-  geocoder.geocode({ address }, (results, status) => {
-    if (status !== OK) {
-      callback({ status }, null, results)
-      return
-    }
+    geocoder.geocode({ address }, (results, status) => {
+      if (status !== OK) {
+        if (callback) {
```

Comment:
@jxom Let's warn the user using this library with `console.warn`. Something like

```js
console.warn('Callback for geocodeByAddress is deprecated and will be removed in the next version on react-places-autocomplete. Please see: https://github.com/kenny-hibino/react-places-autocomplete/blob/master/README.md#geocodebyaddress-api')
```